config/routes.rb

```diff
@@ -449,6 +449,11 @@
     resources :participants, only: %i[index show]
     resources :participant_profiles, only: [] do
       namespace :ecf do
+        resource :participant_declarations, only: [] do
+          get ":participant_declaration_id/void/new", to: "participant_declarations/void#new", as: :new_void
+          post ":participant_declaration_id/void", to: "participant_declarations/void#create", as: :create_void
+        end
```

Comment:
Should we follow a more restful route?
```ruby
namespace :ecf do
  resource :participant_declarations, only: [] do
    resources :voids, only: [:new, :create]
  end
end
```

Hmm, maybe we should rename controller to "MakeVoidsController"

I suppose we could also do it with singular resource:
```ruby
    resource :void, only: [:new, :create]
```

Reply:
Yeah I agree the route structure isn't ideal; I opted for consistency here though to keep it the same as the change training status action